src/repl.zig

```diff
@@ -770,10 +729,10 @@ pub fn ReplType(comptime MessageBus: type) type {
                         // Re-fill the buffer with prefix, current match and suffix. Set the
                         // buffer index where the current selected match ends.
                         repl.buffer.clearRetainingCapacity();
-                        try repl.buffer.appendSlice(match_prefix.items);
-                        try repl.buffer.appendSlice(current_match);
-                        try repl.buffer.appendSlice(match_suffix.items);
-                        buffer_index = repl.buffer.items.len - match_suffix.items.len;
+                        try repl.buffer.appendSlice(repl.completion.prefix.slice());
+                        try repl.buffer.appendSlice(current_completion);
+                        try repl.buffer.appendSlice(repl.completion.suffix.slice());
```

Comment:
Thanks for pointing out this issue. I have added a check early in the tab completion code to skip the current completion if adding it would overflow the buffer.